### PR TITLE
[Dependency Analysis] Add Android Gradle Plugin

### PR DESCRIPTION
**Pull request by ParaskP7**
Project Thread: paaHJt-6yU-p2
Required By: [BuildkiteCI#473](https://github.com/Automattic/buildkite-ci/pull/473)
Depends On:
- [x] #146

This PR adds the dependency analysis Android Gradle plugin to this project, for dependency analysis purposes.

### Description

For now, only the main `buildHealth` task is going to be utilized and produce data once every week on CI (see this [commit](https://github.com/wordpress-mobile/WordPress-Utils-Android/pull/148/commits/bed6aeb4e85ac1a19e3be43bad6778ac1a117caf) and [this](https://github.com/Automattic/buildkite-ci/pull/473) PR). Amongst other, this data will include the following:

- Number of modules (`projectCount`)
- Unused dependencies which should be removed (`unusedCount`)
- Transitive dependencies which should be declared directly (`undeclaredCount`)
- Existing dependencies which should be modified to be as indicated (`misDeclaredCount`)
- Dependencies which could be compile-only (`compileOnlyCount`)
- Dependencies which should be removed or changed to runtime-only (`runtimeOnlyCount`)

Afterward, this data will get collected from CI and uploaded to our Apps Metrics infrastructure, for visualization and alerting purposes.

### Testing Steps

1. Local: Run the `./gradlew buildHealth` task and verify that under the root level `build/reports/dependency-analysis` folder you get the below 2 reports both, in JSON and text format:
    - `build-health-report.json`
    - `build-health-report.txt`
2. CI: Using the `New Build` 🟢 CI button for [UtilsAndroid](https://buildkite.com/automattic/wordpress-utils-android), test this standalone dependency analysis job (see form below). Then:
    - When CI starts, make sure that adding `/meta-data` to that CI build's URL ([example](https://buildkite.com/automattic/wordpress-utils-android/builds/246/meta-data)) will give you one extra meta-data, the `pipeline_file` one, with a value of `schedules/dependency-analysis.yml` (see screenshot below):
    - When CI completes, make sure that the below 2 CI artifacts are available both, in JSON and text format:
        - `build/reports/dependency-analysis/build-health-report.json`
        - `build/reports/dependency-analysis/build-health-report.txt`
3. REST: Using the Buildkite's REST API you could query for this type of builds, using the [meta-data](https://buildkite.com/docs/apis/rest-api/builds#list-all-builds) query parameter, and see that the result is the expect one. Use this query: `https://api.buildkite.com/v2/organizations/automattic/pipelines/wordpress-utils-android/builds?meta_data[pipeline_file]=schedules/dependency-analysis.yml`

image

image

### CODE REVIEW

File: .buildkite/schedules/dependency-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/buildkite/pipeline-schema/main/schema.json
+---
+
+common_params:
+  # Common plugin settings to use with the `plugins` key.
+  - &common_plugins
+    - automattic/a8c-ci-toolkit#2.15.0
```
Comment (wzieba) [inline .buildkite/schedules/dependency-analysis.yml:7]: Can we use the newest version?
```suggestion
    - automattic/a8c-ci-toolkit#3.4.2
```

**Review by wzieba — APPROVED** [review]

File: .buildkite/schedules/dependency-analysis.yml (hunk 1 above)
Comment (ParaskP7) [reply .buildkite/schedules/dependency-analysis.yml:7]: I would love to do that @wzieba but I really don't want to make a change that can potentially block this PR (ie. failing CI, etc), me not knowing exactly what the diff between the 2 version are. 🤷 

Can I maybe do that on a separate PR next week, and do that for all these 6 libraries I have been (Login, Media Picker) and will be touching next week (Aztec, About, Tracks)? 🙏

Comment (wzieba) [reply .buildkite/schedules/dependency-analysis.yml:7]: What works best for you - the only breaking difference is related `nvm` installation, but if you prefer to do this in a separate PR, it's ok to me too 👍

Comment (ParaskP7) [reply .buildkite/schedules/dependency-analysis.yml:7]: Cool, great, I've noted this down and will create separate PRs for each library I have been or will be touching next week. 🗒️ 

> the only breaking difference is related nvm installation

👍

Comment (ParaskP7) [reply .buildkite/schedules/dependency-analysis.yml:7]: FYI @wzieba that this is now done: #150

**Pull request merged**